tests/test_server.py

```diff
@@ -1067,3 +1068,268 @@ async def execute_resource_deploy_start(expected_return_code: int, resulting_nr_
+                tid=env_id,
+                resource_id=rvid_r1_v1,
+                action_id=action_id,
+                status=const.ResourceState.deployed,
+                messages=[
+                    LogLine(level=const.LogLevel.DEBUG, msg="message", kwargs={"keyword": 123}, timestamp=now),
+                    LogLine(level=const.LogLevel.INFO, msg="test", kwargs={}, timestamp=now),
+                ],
+                changes={"attr1": AttributeStateChange(current=None, desired="test")},
+                change=const.Change.purged,
+                send_events=True,
+            )
+            assert result.code == 200, result.result
+        else:
+            result = await agent._client.resource_action_update(
+                tid=env_id,
+                resource_ids=[rvid_r1_v1],
+                action_id=action_id,
+                action=const.ResourceAction.deploy,
+                started=None,
+                finished=now,
+                status=const.ResourceState.deployed,
+                messages=[
+                    data.LogLine.log(level=const.LogLevel.DEBUG, msg="message", timestamp=now, keyword=123),
+                    data.LogLine.log(level=const.LogLevel.INFO, msg="test", timestamp=now),
+                ],
+                changes={rvid_r1_v1: {"attr1": AttributeStateChange(current=None, desired="test")}},
+                change=const.Change.purged,
+                send_events=True,
+            )
+            assert result.code == 200, result.result
+
+    # Assert effect of resource_deploy_done call
+    assert f"{rvid_r1_v1}: message" in caplog.messages
+    assert f"{rvid_r1_v1}: test" in caplog.messages
+
+    result = await client.get_resource_actions(tid=env_id)
+    assert result.code == 200, result.result
+    assert len(result.result["data"]) == 1
+    resource_action = result.result["data"][0]
+    assert resource_action["environment"] == str(env_id)
+    assert resource_action["version"] == model_version
+    assert resource_action["resource_version_ids"] == [rvid_r1_v1]
+    assert resource_action["action_id"] == str(action_id)
+    assert resource_action["action"] == const.ResourceAction.deploy
+    assert resource_action["started"] is not None
+    assert resource_action["finished"] is not None
+    assert resource_action["messages"] == [
+        {
+            "level": const.LogLevel.DEBUG.value if endpoint_to_use == "resource_deploy_done" else const.LogLevel.DEBUG.name,
+            "msg": "message",
+            "args": [],
+            "kwargs": {"keyword": 123},
+            "timestamp": now.isoformat(timespec="microseconds"),
+        },
+        {
+            "level": const.LogLevel.INFO.value if endpoint_to_use == "resource_deploy_done" else const.LogLevel.INFO.name,
+            "msg": "test",
+            "args": [],
+            "kwargs": {},
+            "timestamp": now.isoformat(timespec="microseconds"),
+        },
+    ]
```

Comment:
I am not happy about the inconsistency in the representation of the `level` parameter in the JSON format.

* In the `resource_action_update` endpoint the `LogLevel` enum is not part of a pydantic object. As such, it is serialized using the name of the enum (See: https://github.com/inmanta/inmanta-core/blob/75fecbedb13e2f78f36385d88ff4ea1681384e0a/src/inmanta/util.py#L274)
* In the `resource_deploy_done` endpoint, the `LogLevel` enum is part of the pydantic object `LogLine`. As such, it serializes it using the value, because the `use_enum_values` config option is set (See: https://github.com/inmanta/inmanta-core/blob/75fecbedb13e2f78f36385d88ff4ea1681384e0a/src/inmanta/data/model.py#L44)

Any opinions on this?

Reply:
I think we best keep this backward compatible. Can we override this setting for this specific field?

Reply:
Making it backwards compatible would be an inconsistency too (not saying we shouldn't). The pydantic models in `inmanta.data.model` are generally usable over the API boundary. If we make this use enum names, `LogLine(**json.encode(log_line.dict(), default=api_boundary_json_encoder))` will not work anymore because pydantic expects enum values as input.

Reply:
Yes. We can. In that case everything serialises correctly to the name of the enum. The problem is that a validation error occurs because pydantic can only instantiate pydantic objects from the value of an enum.

Reply:
Though we could add a custom pre-validator to convert any enum names. It still feels a bit weird but I think it would be preferable to the inconsistency with the other endpoint. We should add a clear comment if we do this.